### PR TITLE
Fixes #12393

### LINKED ISSUE #12393: Improve testsuite for cookie authentication
Currently the cookie authentication tests are quite poorly matching the functionality of the code. There should be separate tests for each authentication step (login, recall from cookies, password change and logout) for possible configurations (plain, arbitrary server or captcha). Some of these tests already exists, but they probably need cleanup as the underlaying code has been changed a lot in 4.6.4.

### PR DESCRIPTION
**Pull request by osaid-r**
Signed-off-by: osaid <osaid.nasir@gmail.com>

Before submitting pull request, please check that every commit:

- [X] Has proper Signed-Off-By
- [X] Has commit message which describes it
- [X] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [X] Any new functionality is covered by tests

### CODE REVIEW

File: test/classes/plugin/auth/AuthenticationCookieTest.php [hunk 1]
```diff
@@ -1049,6 +1049,40 @@ public function testAESSecretSplit($secret, $mac, $aes)
...
+            'server' => 'b 2'
+        );
+        $method = new \ReflectionMethod(
+            'PMA\libraries\plugins\auth\AuthenticationCookie',
+            '_getSessionEncryptionSecret'
+        );
+        $method->setAccessible(true);
+
+        $result = $method->invoke($this->object, null);
+        $GLOBALS['PMA_Config']->setCookie(
```
Comment (ragnerok) [inline test/classes/plugin/auth/AuthenticationCookieTest.php]: Yeah but then I'll have to use assertContains() right ?

Comment (nijel) [inline test/classes/plugin/auth/AuthenticationCookieTest.php]: You just need to compare whether cookieEncrypt return value is same as the pmaAuth cookie, or am I missing something?

Comment (ragnerok) [inline test/classes/plugin/auth/AuthenticationCookieTest.php]: So you are saying I should simply assertEquals pmaAuthCookie with returned value of cookieEncrypt but I was saying that I could call cookiedecrypt and assert with$newPassword

Comment (nijel) [inline test/classes/plugin/auth/AuthenticationCookieTest.php]: The cookieDecrypt takes string as a parameter, there is no need to pollute $_COOKIES superglobal for the test purpose here.

Comment (ragnerok) [inline test/classes/plugin/auth/AuthenticationCookieTest.php]: Yeah ok

**Comment by ragnerok** [comment]
I think that the build is failing because createIV() probably returns random strings to cookieEncrypt(), right ?

**Comment by nijel** [comment]
Yes, createIV is supposed to return new IV every time.

**Comment by ragnerok** [comment]
So if I set $_cookie_iv then this problem shouldn't be there

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**